gateway/handlers/infohandler.go

```diff
@@ -0,0 +1,56 @@
+package handlers
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"github.com/openfaas/faas/gateway/types"
+	"github.com/openfaas/faas/gateway/version"
+)
+
+// MakeInfoHandler is responsible for display component version information
+func MakeInfoHandler(h http.Handler) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		sw := types.NewStringResponseWriter()
+		h.ServeHTTP(sw, r)
+
+		log.Printf("Body: %s", sw.Body())
+		provider := make(map[string]interface{})
+		providerVersion := &types.VersionInfo{}
+
+		err := json.Unmarshal(sw.Body(), &provider)
+		if err != nil {
+			log.Printf("Error unmarshalling provider json. Got %s. Error %s\n", string(sw.Body()), err.Error())
+		}
+
+		versionMap := provider["version"].(map[string]interface{})
```

Comment:
Perhaps we could use structs instead of interface{}, what do you think?

Reply:
I wanted to avoid adding a dependency to `vendor/github.com/openfaas/faas-provider/types` where the `InfoRequest` type is defined. So on line [19](https://github.com/openfaas/faas/pull/737/files/32411682d71740f4ffe36359888ef0ff26e0836c#diff-2a00b38d47e23a0bd2ab56523a978337R19) I used `map[string]interface{}`. And the whole object is now a graph of `map[string]interface{}`.